src/fxcorrelator.py

```diff
@@ -201,20 +325,32 @@ def setup_sensors(self, katcp_server):
         """
         if not self._initialised:
             raise RuntimeError('Cannot set up sensors until instrument is initialised.')
+        ioloop = getattr(self, 'ioloop', None)
+        if not ioloop:
+            ioloop = getattr(katcp_server, 'ioloop', None)
+        if not ioloop:
+            # make my own ioloop
+            ioloop = IOLoop.instance()
+            ioloop.start()
```

Comment:
You need to start the ioloop in another thread. ioloop.start() will block, which means your program would go nowhere!  I.e. something like:

```py
self.ioloop = IOLoop.instance()
def run_ioloop():
    try:
        self._ioloop.start()
        self._ioloop.close()
    except Exception:
        self._logger.exception('Error running tornado IOloop: ')
    finally:
        self.ioloop_thread =  None
        self.ioloop = None

t = self.ioloop_thread = threading.Thread(target=run_ioloop)
t.setDaemon(True)  # Or make it configurable?
t.start()
```

You might also consider moving this whole ioloop-getting business (i.e all the code from ioloop = getattr(self, 'ioloop', None) ... till after starting the thread) into a separate method (get_ioloop()?) to make the code a bit cleaner.